modules/mysql/main.tf

```diff
@@ -94,6 +95,12 @@ resource "google_sql_database_instance" "default" {
         record_client_address   = lookup(insights_config.value, "record_client_address", false)
       }
     }
+    dynamic "data_cache_config" {
```

Comment:
@ibhaskar2 plz make following change otherwise tests will fail. This value is immutable only for ENTERPRISE_PLUS tier and when setting value to true. Otherwise it will keep updating the value. 

```
    dynamic "data_cache_config" {
      for_each = var.edition == "ENTERPRISE_PLUS" && var.data_cache_enabled ? ["cache_enabled"] : []
      content {
        data_cache_enabled = var.data_cache_enabled
      }
    }
```